app/Main.hs

```diff
@@ -171,6 +191,9 @@ parseOptions =
               )
           )
 
+labelParser :: Text.Text -> (Rule.LabelName, Rule.LabelType)
```

Comment:
Current behavior is to silently ignore that and behave as if the user had specified `text` as the label type. I agree that this is not ideal.
 Let me try out how hard it is to implement parsing the label type. I'll try to combine it with making the label type a real custom data type instead of a alias to `Text.Text`.

Reply:
This was finicky, but not too bad to implement.
Although yaml file do not support this type of custom data type very well, so the parser for the config file essentially abuses the yaml spec a bit, but the worst problem with that is that the yaml parser does not generate a fool proof error message, when parsing fails.
I've deliberately kept the behavior that not specifying a label type at all and specifying `text` is the same (i.e. no restriction on the content of the label), because this gives us a neat way to just require the presence of a label on the command line:
```
$ hadolint --require-label foo Dockerfile
```